lib/Pool.js

```diff
@@ -28,6 +28,13 @@ const Deferred = require("./Deferred");
  *   When the pool is created, or a resource destroyed, this minimum will
  *   be checked. If the pool resource count is below the minimum, a new
  *   resource will be created and added to the pool.
+ * @param {Number} [factory.maxUses=0]
+ *   The number of times an item is to be used before it is destroyed
+ *   no matter whether it is still healthy.  A value of 0 indicates the
+ *   item should be used indefinitely so long as it is healthy.
+ *   This can help with "re-balancing" connections when pool members behind
+ *   a load balancer are added but are not being adopted due to pools being
+ *   full of pre-existing persistent connections.
```

Comment:
How about using `Infinity` instead of 0? It works, is easier to code (no special case to consider), and it is also clearer for the programmer in my point of view.

Reply:
That's an interesting idea.  I will try it out.

Reply:
I switched to a default of Infinity and simplified the condition check below.